docs/admin/migrating_to_v3.md

```diff
@@ -0,0 +1,151 @@
+    - [I’m seeing a banner with the message \`Some Device Lifecycle Management models have not been migrated to Nautobot core models…\`](#im-seeing-a-banner-with-the-message-some-device-lifecycle-management-models-have-not-been-migrated-to-nautobot-core-models)
+    - [This job is taking a long time.](#this-job-is-taking-a-long-time)
+
+  
+
+# Migrating to DLM app version 3.0
+
+**Device Lifecycle Management App** version 3.0 now supports these core Nautobot models: SoftwareVersion, SoftwareImageFile, and Contact.
+
+These models were introduced in Nautobot 2.2.0 as functional equivalents to the now deprecated DLM models. **DLM models will be removed in version 4.2**. The table below shows the corresponding model in core that matches each DLM model.
+
+| Core model | DLM model |
+| :---- | :---- |
+| SoftwareVersion | SoftwareLCM |
+| SoftwareImageFile | SoftwareImageLCM |
+| Contact | ContactLCM |
+
+The DLM models and their instances will remain in place to ensure a smooth migration and prevent data loss.
+
+After installing app version 3.0, all existing instances of these models must be moved to the core models. This will enable full DLM app functionality. The migration process is outlined in the following sections.
+
+!!! warning
+    Ensure that the DLM app is at least version 2.0.0 and Nautobot is version 2.2.0 or later before starting migration.
+
+## Process of Migration
+
+Before continuing to use the app, it is essential to run the **Device Lifecycle Management to Nautobot Core Model Migration** job, which was introduced in version 3.0. This job migrates existing SoftwareLCM, SoftwareImageLCM, and ContactLCM models to the corresponding core models. Importantly, this job can be run multiple times, allowing existing automation or workflows to continue targeting deprecated models and providing operators with additional control over the update process.
+
+The DLM objects will be marked as migrated by setting a boolean value to true after the job creates instances of core objects that match them and records a reference to the core object on the old DLM object during the migration. The deprecated DLM models have been updated with the attributes below to allow for migration tracking:
+
+* **migrated\_to\_core\_model** \- Foreign Key to the core model instance  
+* **migrated\_to\_core\_model\_flag** \- Boolean set to True when the instance is migrated to the core
+
+The migration job generates new core objects that replace the deprecated DLM objects. If the job encounters an error, it will not rollback any created objects. If an error occurs, check the job logs to identify and resolve the issue, then restart the job. The job will revalidate any objects that have already been migrated and retry the migration for those objects that failed during previous runs.
```

Comment:
Would this be accurate?

```suggestion
The migration job generates new core objects that replace the deprecated DLM objects. If a core object already exists that matches the existing dlm object, that may be used instead. If the job encounters an error, it will not rollback any created objects. If an error occurs, check the job logs to identify and resolve the issue, then restart the job. The job will revalidate any objects that have already been migrated and retry the migration for those objects that failed during previous runs.
```

Reply:
Yes, to both, I'll update as per the suggestion and will add a sentence on the dryrun mode.